src/Microsoft.AspNetCore.Testing/xunit/ConditionalTheoryDiscoverer.cs

```diff
@@ -14,6 +15,24 @@ public ConditionalTheoryDiscoverer(IMessageSink diagnosticMessageSink)
         {
         }
 
+        public override IEnumerable<IXunitTestCase> Discover(ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo theoryAttribute)
+        {
+            var testCases = base.Discover(discoveryOptions, testMethod, theoryAttribute);
+            // Xunit evaluates MemeberData before skip conditions. If there is no member data it returns an error test case.
+            // However we have some cases where we conditionally expect no MemeberData and we set skip conditions accordingly.
+            // E.g. All of the test cases are excluded on a specific OS. In other words, don't fail if it was going to be skipped anyways.
+            // We don't always do this because we want to show the individual test cases as skipped where possible.
+            if (testCases.Count() == 1 && testCases.First() is ExecutionErrorTestCase)
```

Comment:
Assert error message too there are different issues that can cause ExecutionErrorTestCase.